src/sql/workbench/services/fileBrowser/browser/urlBrowserDialog.ts

```diff
@@ -0,0 +1,477 @@
+		this._backupFileSelectorBox.setOptions(backupFilesDisplayNames);
+		this._backupFileSelectorBox.select(0);
+		if (this._backupFiles.length === 0) {
+			this._backupFileSelectorBox.disable();
+		} else {
+			this._backupFileSelectorBox.enable();
+		}
+	}
+
+	private setBackupFilesSelectorError(errors: any) {
+		this._backupFileSelectorBox.setOptions([]);
+		this._backupFileSelectorBox.select(0);
+		this._backupFileSelectorBox.disable();
+	}
+
+	private setBackupFileDefaultValue() {
+		this._backupFileInputBox.value = this._defaultBackupName;
+	}
+
+	public open(ownerUri: string,
+		expandPath: string,
+		fileFilters: [{ label: string, filters: string[] }],
+		fileValidationServiceType: string,
+	): void {
+		this._ownerUri = ownerUri;
+		this.enableOkButton();
+		this.enableCreateCredentialsButton();
+		this.spinner = true;
+		this.show();
+		this.spinner = false;
+	}
+
+	/* enter key */
+	protected override onAccept() {
+		let selectedValue = this._sasInputBox.value;
+		if (this._okButton.enabled === true && selectedValue !== '') {
+			this.ok();
+		}
+	}
+
+
+	private enableOkButton() {
+		if (strings.isFalsyOrWhitespace(this._blobContainerSelectorBox.value) || strings.isFalsyOrWhitespace(this._sasInputBox.value) || (this._restoreDialog && strings.isFalsyOrWhitespace(this._blobContainerSelectorBox.value)) || (!this._restoreDialog && strings.isFalsyOrWhitespace(this._backupFileInputBox.value))) {
+			this._okButton.enabled = false;
+		} else {
+			this._okButton.enabled = true;
+		}
+	}
+
+	private enableCreateCredentialsButton() {
+		if (strings.isFalsyOrWhitespace(this._blobContainerSelectorBox.label)) {
+			this._sasButton.enabled = false;
+		} else {
+			this._sasButton.enabled = true;
+		}
+	}
+
+	private ok() {
+		let returnValue = '';
+		if (this._restoreDialog) {
+			returnValue = `https://${this._storageAccountSelectorBox.value}.blob${this._selectedAccount.properties.providerSettings.settings.azureStorageResource.endpointSuffix}/${this._blobContainerSelectorBox.value}/${this._backupFileSelectorBox.value}`;
+		} else {
+			returnValue = `https://${this._storageAccountSelectorBox.value}.blob${this._selectedAccount.properties.providerSettings.settings.azureStorageResource.endpointSuffix}/${this._blobContainerSelectorBox.value}/${this._backupFileInputBox.value}`;
+		}
+		this._onOk.fire(returnValue);
+		this.close('ok');
+	}
+
+
+	private close(hideReason: HideReason = 'close'): void {
+		this._onOk.dispose();
+		this.hide(hideReason);
+	}
+
+	private generateSharedAccessSignature() {
+		this.spinner = true;
+		const blobContainerUri = `https://${this._storageAccountSelectorBox.value}.blob${this._selectedAccount.properties.providerSettings.settings.azureStorageResource.endpointSuffix}/${this._blobContainerSelectorBox.value}`;
+		this._azureAccountService.getStorageAccountAccessKey(this._selectedAccount, this._selectedSubscription, this._selectedStorageAccount)
+			.then(getStorageAccountAccessKeyResult => {
```

Comment:
Instead of using nested .thens it would be much more preferable to make this function async (and then the caller could choose to await it or not)